### ISSUE TITLE
default entry file to `.`

### ISSUE THREAD

**Issue by jasonkarns** [substack/node-browserify#1378]
Presently, browserify (CLI) requires an entry file to start as the entrypoint.

```
$ browserify
Usage ...
```

Of course, the entrypoint file is typically specified via `browserify index.js`. In package.json, a typical build script might look like:
```
"build": "browserify $npm_package_main"
```

Apparently, since node already knows how to resolve the default file in a directory, `.` can be supplied to browserify as the entry as well:

```
"build": "browserify ."
```

Since node (and apparently browserify, as well) can accept a directory as an entrypoint; it would be nice if browserify simply defaulted to `.` as the entrypoint if not specified via the command line.

I would love my build script to be:

```
"build": "browserify -o $npm_package_config_buildfile"
```

**Comment by chielkunkels** [comment]
I don't think that's really a possibility since an entry file is not always desired. In my project, for example, I have some bundles without an entry point but only `.require()`'d modules. I do this cause I lazy-load them and then `require` them in my javascript file in order to access the contents.

**Comment by jasonkarns** [comment]
That's probably not the most common use case for browserify, right? And, I'd imagine, would still work (even with this feature) if there simply isn't an index.js (or `main`).

For cases where a bundle of modules (without an entrypoint) is the desired outcome, and `main`/index.js exists, a flag could be used.

I'm suggesting a flag be used to force the entrypoint to be ignored because I'm wagering that using browserify with an entrypoint of `main`/index.js is the most common 80% use case. So a sane default for the common case is desirable; with a flag for the edge case.

**Comment by chielkunkels** [comment]
Well, the thing is that an entry point is executed immediately, a require has to be required and then it can do stuff. One thing I like a lot about browserify is that, with some configuration, you can get it to do anything you want.

I don't really understand why you'd want to default to `.`, I personally rarely keep my code there, usually it's stuffed away in `src` or similar. I think that's the main reason there is no default, because it's opinionated. And if there's something browserify is not, it is opinionated.

Just my two cents.

**Comment by jasonkarns** [comment]
> I personally rarely keep my code there, usually it's stuffed away in src or similar.

It doesn't matter where the source is; an entrypoint of `.` respects the package.json `main` field. If `main` is `lib/my-module/foo.js` then that's what gets required as `.`.

**Comment by chielkunkels** [comment]
Fair enough. I still think it's very little effort to specify the entry file though, but to each their own. :)

**Comment by jasonkarns** [comment]
It is :)

I'm only falling back to the trite "sane defaults"
